packages/@sanity/desk-tool/src/utils/createPaneResolver.ts

```diff
@@ -53,37 +53,49 @@ const rethrowWithPaneResolutionErrors: PaneResolverMiddleware = (next) => (
   }
 }
 
+const wrapWithPublishReplay: PaneResolverMiddleware = (next) => (...args) => {
+  return next(...args).pipe(
+    // need to add publishReplay + refCount to ensure new subscribers always
+    // get an emission. without this, memoized observables may get stuck
+    // waiting for their first emissions resulting in a loading pane
+    publishReplay(1),
+    refCount()
```

Comment:
Not 100% how it fits into the broader context, but did you consider using `shareReplay` here? If you use `shareReplay({bufferSize: 1, refCount: true})` instead of `publishReplay(1)` & `refCount()` the stream will be retry-able in case of an error. (`publishReplay` will replay any error).

From the [shareReplay](https://rxjs.dev/api/operators/shareReplay) docs:
> A successfully completed source will stay cached in the shareReplayed observable forever, but an errored source can be retried.

Reply:
ah that's a good thought. i did not consider it.

thinking about it now, i'd say the desired behavior is to replay the error. e.g. if the source errors once then switching panes should replay the error i'd think